### ISSUE TITLE
Java errors not reported in signs or diagnostics

### ISSUE THREAD

**Issue by bgx90** [prabirshrestha/vim-lsp#366]
I have configured vim-lsp to use jdt.ls for Java as shown in the vim-lsp wiki.  Using vim 8.1 and neovim 0.3.4 I am seeing warnings in the signs column and the document diagnostics as I would expect, but when I introduce an error into the code, it is not flagged or reported and any warning that was reported following the location of the error disappears.  I have also tried using the latest snapshot of jdt.ls (jdt-language-server-0.37.0-201904162228), but it does not resolve the problem.  Using jdt.ls with neoclide/coc and neoclide/coc-java, I see the error reported as I would expect, so it seems that vim-lsp is not able to show Java errors from jdt.ls.

What vim-lsp shows before I introduce an error:
![image](https://user-images.githubusercontent.com/7351168/56617517-b2037d00-65e5-11e9-865b-d28e9d3674f8.png)

What vim-lsp shows after I introduce an error on line 8 (the error is not reported):
![image](https://user-images.githubusercontent.com/7351168/56617595-f000a100-65e5-11e9-9121-3ee5178b9884.png)

What neoclide/coc shows for the same error on line 8 (the error is reported correctly):
![image](https://user-images.githubusercontent.com/7351168/56617838-7cab5f00-65e6-11e9-9cdf-59ddc8426161.png)

**Comment by tsufeki** [comment]
I could reproduce the error, in vim-lsp logs I've found:

    Classpath is incomplete. Only syntax errors will be reported for file:///home/...

According to [this page](https://github.com/redhat-developer/vscode-java/wiki/%22Classpath-is-incomplete%22-warning), you need a proper java project directory (for example, with `pom.xml`).

**Comment by bgx90** [comment]
I confirmed that error reporting from vim-lsp works as expected with maven projects.  I think the difference in behavior between vim-lsp and neoclide/coc-java is that coc-java sends the current working directory to the language server as the project root when a project descriptor like a pom does not exist.  The server then gives an error complaining about an invalid project description in that case, but the error checking still works as expected.  Since coc-java is based on the vscode-java extension @tsufeki referenced, I'm surprised that vscode-java does not do the same.  I checked the vscode-java behavior and it behaves like lsp-vim and does not show the error.  If vim-lsp's goal is to match the behavior of the vscode-java extension, then I suppose it is working as the developer intended.  Error checking of standalone java files would be nice to have though, so maybe an enhancement to do what coc-java does could be added in the future.
